### PR TITLE
[CLI] Fix genesis client yaml file on Windows

### LINKED ISSUE #10371: Local RPC configuration of http://0.0.0.0:9000 is not working for Windows
## Steps to Reproduce Issue

1. Install sui on Windows
2. Start a local node 
3. Try to run a cli command, eg `sui client gas`

## Expected Result

The gas coin list

## Actual Result

```
PS C:\Users\teoha> sui client gas
Networking or low-level protocol error: HTTP error: error trying to connect: tcp connect error: The requested address is not valid in its context. (os error 10049)

Caused by:
    HTTP error: error trying to connect: tcp connect error: The requested address is not valid in its context. (os error 10049)

Stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: <unknown>
  14: <unknown>
  15: <unknown>
```

if client.yaml file is changed to 
```
  - alias: localnet
    rpc: "http://127.0.0.1:9000"
```

commands work fine:
```
PS C:\Users\teoha> sui client gas
                             Object ID                              |  Gas Value
----------------------------------------------------------------------------------
 0x774f93c3823ea31d1265f53f604c9ee1a370c70523ead28c33ffeb742ae6b806 | 30000000000000000
 0xb9dbbe8c96e0a0a11d12347223615e228bcf1f4bed42d7c1826d29f5dc009e3e | 30000000000000000
 0xd803cb790861dfcd7cc606b3878f40465fa7317253be312b909b9a7682737746 | 30000000000000000
 
```

### PR DESCRIPTION
**Pull request by stefan-mysten**
## Description 
On Windows, binding the localnet IP address to 0.0.0.0 leads to issues and failing with a networking error for any CLI command when the localnet network is the default one. By default, when starting a Sui local network using `sui start` and when there is no default configuration, the command will generate a genesis including a `client.yaml` file that contains the localnet IP address set to `0.0.0.0`. This PR fixes that for the `client.yaml` file and should close #10371.

## Test plan 

On a Windows box, run the binary and tested that it can connect to the network.

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [x] CLI: When `sui start` is called without a sui config directory, a genesis will be generated along with all the config files, including the client.yaml file that creates a localnet with IP set to 0.0.0.0, which leads to problems on Windows. This change fixes that and sets the IP to be 127.0.0.1. This is also the case when `sui genesis` is used.
- [ ] Rust SDK:

### CODE REVIEW

File: crates/sui/src/sui_commands.rs [hunk 1]
```diff
@@ -1011,9 +1012,22 @@ async fn genesis(
     if client_config.active_address.is_none() {
         client_config.active_address = active_address;
     }
+
+    // On windows, using 0.0.0.0 will usually yield in an networking error. This localnet ip
+    // address must bind to 127.0.0.1 if the default 0.0.0.0 is used.
+    let localnet_ip =
+        if fullnode_config.json_rpc_address.ip() == IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)) {
+            "127.0.0.1".to_string()
```
Comment (amnn) [inline crates/sui/src/sui_commands.rs:1020]: Is there somewhere we can change this more uniformly so that we don't introduce `0.0.0.0` in the config anymore?

**Review by amnn — APPROVED** [review]

File: crates/sui/src/sui_commands.rs (hunk 1 above)
Comment (stefan-mysten) [reply crates/sui/src/sui_commands.rs:1020]: This is set in the genesis function, here. I don't like the solution either, but I do not know if I will break anything if I change the JSON RPC address there.

https://github.com/MystenLabs/sui/blob/main/crates/sui/src/sui_commands.rs#L960

Comment (amnn) [reply crates/sui/src/sui_commands.rs:1020]: Yes, I think this does have broader implications: https://stackoverflow.com/questions/20778771/what-is-the-difference-between-0-0-0-0-127-0-0-1-and-localhost

TL;DR: `0.0.0.0` means accept traffic from all interfaces, whereas `127.0.0.1` means accept traffic from the loopback interface only (only traffic from the same host), conventionally.

This makes me wonder whether this is the right move even for Windows?

Comment (amnn) [reply crates/sui/src/sui_commands.rs:1020]: Actually, maybe this is fine, and it suggests that changing it during environment creation is the right thing to do: We're saying that the config during genesis (which tells the network where to accept connections from) should accept from any interface, but when we are making the connection, we choose to make it via the loopback -- that seems fine.
